app/controllers/foreign/income/ForeignPropertyRentalIncomeController.scala

```diff
@@ -30,44 +30,55 @@ import views.html.foreign.income.ForeignPropertyRentalIncomeView
-                                                       val controllerComponents: MessagesControllerComponents,
-                                                       view: ForeignPropertyRentalIncomeView
-                                                     )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+class ForeignPropertyRentalIncomeController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  navigator: ForeignPropertyNavigator,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: ForeignPropertyRentalIncomeFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: ForeignPropertyRentalIncomeView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController with I18nSupport {
 
-
-
-  def onPageLoad(taxYear: Int,  mode: Mode,countryCode:String): Action[AnyContent] = (identify andThen getData andThen requireData) {
-    implicit request =>
+  def onPageLoad(taxYear: Int, mode: Mode, countryCode: String): Action[AnyContent] =
+    (identify andThen getData andThen requireData) { implicit request =>
       val form = formProvider(request.user.isAgentMessageKey)
       val preparedForm = request.userAnswers.get(ForeignPropertyRentalIncomePage(countryCode)) match {
-        case None => form
+        case None        => form
         case Some(value) => form.fill(value)
       }
 
       Ok(view(preparedForm, taxYear, request.user.isAgentMessageKey, mode, countryCode))
-  }
+    }
 
-  def onSubmit(taxYear: Int, mode: Mode, countryCode:String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
-    implicit request =>
+  def onSubmit(taxYear: Int, mode: Mode, countryCode: String): Action[AnyContent] =
+    (identify andThen getData andThen requireData).async { implicit request =>
       val form = formProvider(request.user.isAgentMessageKey)
-      form.bindFromRequest().fold(
-        formWithErrors =>
-          Future.successful(BadRequest(view(formWithErrors, taxYear, request.user.isAgentMessageKey, mode,countryCode))),
-
-        value =>
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(ForeignPropertyRentalIncomePage(countryCode), value))
-            _              <- sessionRepository.set(updatedAnswers)
-          } yield Redirect(navigator.nextPage(ForeignPropertyRentalIncomePage(countryCode), taxYear, mode, request.userAnswers, updatedAnswers))
-      )
-  }
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors =>
+            Future
+              .successful(BadRequest(view(formWithErrors, taxYear, request.user.isAgentMessageKey, mode, countryCode))),
+          value =>
+            for {
+              updatedAnswers <-
+                Future.fromTry(request.userAnswers.set(ForeignPropertyRentalIncomePage(countryCode), value))
```

Comment:
completed section saving in the wrong path

<img width="564" alt="Screenshot 2024-12-06 at 11 19 47" src="https://github.com/user-attachments/assets/d15372b9-be6d-4c0b-aab9-c43242518519">

Reply:
Done